src/core/bpf-firewall.c

```diff
@@ -392,9 +430,22 @@ static int bpf_firewall_prepare_access_maps(
                 if (!cc)
                         continue;
 
-                bpf_firewall_count_access_items(verdict == ACCESS_ALLOWED ? cc->ip_address_allow : cc->ip_address_deny, &n_ipv4, &n_ipv6);
+                list = verdict == ACCESS_ALLOWED ? cc->ip_address_allow : cc->ip_address_deny;
+
+                bpf_firewall_count_access_items(list, &n_ipv4, &n_ipv6);
+
+                if (!LIST_IS_EMPTY(list) && n_ipv4 == 1 && n_ipv6 == 1 && !a)
+                        a = ip_address_access_item_is_any(list);
```

Comment:
hmm, is this logic correct? unless i am mistaken we should probably escape the slice loop the instance we find one unit that has ip_address_access_item_is_any() set. there's no need to continue the loop in that case, because we already found that one unit covers the whole IP range.

Reply:
Good point.. with this change we can escape as soon as "any" is found since it encompasses all the other values. And yeah just did some quick tests in some cases checking for n == 1 breaks if something like the current service defines "any" and a parent slice also defines "any".